swarm/api/http/server.go

```diff
@@ -960,24 +1014,38 @@ func (s *Server) HandleGetFile(ctx context.Context, w http.ResponseWriter, r *Re
 		if err != nil {
 			getFileFail.Inc(1)
 			Respond(w, r, err.Error(), http.StatusInternalServerError)
+			sp.Finish()
 			return
 		}
 
 		log.Debug(fmt.Sprintf("Multiple choices! --> %v", list), "ruid", r.ruid)
 		//show a nice page links to available entries
 		ShowMultipleChoices(w, r, list)
+		sp.Finish()
 		return
 	}
 
 	// check the root chunk exists by retrieving the file's size
-	if _, err := reader.Size(nil); err != nil {
+	if _, err := reader.Size(ctx, nil); err != nil {
+		getFileNotFound.Inc(1)
+		Respond(w, r, fmt.Sprintf("file not found %s: %s", r.uri, err), http.StatusNotFound)
+		sp.Finish()
+		return
+	}
+
+	buf, err := ioutil.ReadAll(newBufferedReadSeeker(reader, getFileBufferSize))
```

Comment:
What do you think about keeping this here, so that we can actually measure time spent on nodes, and decouple that from slow clients?